tools/c7n_azure/c7n_azure/resources/public_ip.py

```diff
@@ -32,3 +34,20 @@ class resource_type(ArmResourceManager.resource_type):
             'properties.publicIPAllocationMethod',
             'properties.ipAddress'
         )
+
+
+@PublicIPAddress.action_registry.register('delete')
+class PublicIPDeleteAction(BaseAction):
+
+    schema = type_schema('delete')
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super(PublicIPDeleteAction, self).__init__(data, manager, log_dir)
+        self.client = self.manager.get_client()
+
+    def delete(self, resource_group, public_ip_name):
+        self.client.public_ip_addresses.delete(resource_group, public_ip_name)
```

Comment:
@stefangordon I was wondering the same thing about generic deletion. Is there a way to access the `enum_spec` here? Because right now, the only thing specific to this action is the `public_ip_addresses` property

Reply:
If we can abstract that piece of it out, it should work just fine in the arm class